target/linux/ath79/dts/qca9558_netgear_ex7300.dts

```diff
@@ -0,0 +1,233 @@
+			gpio-controller;
+			#gpio-cells = <2>;
+			registers-number = <1>;
+			lines-initial-states = /bits/ 8 <0xff>;
+			spi-max-frequency = <500000>;
+
+			gpio_latch_bit {
+				gpio-hog;
+				gpios = <4 GPIO_ACTIVE_HIGH>;
+				output-high;
+				line-name = "gpio-latch-bit";
+			};
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		power_green: power_green {
+			label = "netgear:green:power";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+		};
+
+		power_amber: power_amber {
+			label = "netgear:amber:power";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		left_blue {
+			label = "netgear:blue:left";
+			gpios = <&led_gpio 7 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		right_blue {
+			label = "netgear:blue:right";
+			gpios = <&led_gpio 6 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		wps_green {
+			label = "netgear:green:wps";
+			gpios = <&led_gpio 5 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		client_red {
+			label = "netgear:red:client";
+			gpios = <&led_gpio 3 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		client_green {
+			label = "netgear:green:client";
+			gpios = <&led_gpio 2 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		router_red {
+			label = "netgear:red:router";
+			gpios = <&led_gpio 1 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		router_green {
+			label = "netgear:green:router";
+			gpios = <&led_gpio 0 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "WPS button";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		extender_apmode {
+			label = "EXTENDER/APMODE switch";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+			linux,input-type = <EV_SW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&pcie0 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&pll {
+	clocks = <&extosc>;
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot: partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+			};
+
+			caldata: partition@50000 {
+				label = "caldata";
+				reg = <0x050000 0x010000>;
+				read-only;
+			};
+
+			partition@60000 {
+				label = "caldata-backup";
+				reg = <0x060000 0x010000>;
+				read-only;
+			};
+
+			partition@70000 {
+				label = "config";
+				reg = <0x070000 0x010000>;
+			};
+
+			partition@80000 {
+				label = "pot";
+				reg = <0x080000 0x010000>;
+			};
+
+			partition@90000 {
```

Comment:
Please add `compatible`, see
[OpenWrt-Devel] Specifying "firmware" partition format on all DTS targets
http://lists.infradead.org/pipermail/openwrt-devel/2018-November/014706.html
for details.